index.js

```diff
@@ -1,33 +1,212 @@
+        wrapper.requestSingleBib(id, (errorBibReq, results) => {
+        getResult(errorBibReq, results, true, id, operation, currentAttempt)
+            .then(function(entry){
+              resolve (entry);
+            })
+            .catch (function(e) {
+              reject (e);
+            });
+        });
+      } else{
+        console.log('Requesting for item info');
+        var itemIds = [id];
+        wrapper.requestMultiItemBasic(itemIds, (errorItemReq, results) => {
+          getResult(errorItemReq, results, false, itemIds, operation, currentAttempt)
+            .then(function(entry){
+              resolve (entry);
+            })
+            .catch (function(e) {
+              reject (e);
+            });
+        });
+      }
+    });
+  });
+}
 
-// var params = {
-//   StreamName: 'SierraBibPostRequestTest' /* required */
-// };
-// kinesis.describeStream(params, function(err, data) {
-//   if (err) console.log(err, err.stack); // an error occurred
-//   else     console.log(JSON.stringify(data,null,2));           // successful response
+//get bib or item based on prama passed
+var getResult = function(errorResourceReq, results, isBib, resourceId, operation, attemptNumber){
+  return new Promise(function(resolve, reject){
+    if(errorResourceReq){
+        if(errorResourceReq.httpStatus !== null && errorResourceReq.httpStatus === 401){
+          console.log("This is a token issue. Going to renew token");
+          if(isBib) {
+            console.log('Number of attempts made for bib ' + resourceId + ' - ' + attemptNumber);
+          } else{
+            console.log('Number of attempts made for item ' + resourceId + ' - ' + attemptNumber);
+          }
+          set_wrapper_access_token()
+            .then(function (access_token){
+              wrapper_access_token = access_token;
+              if(operation.retry(errorResourceReq)){
+                return;
+                if(isBib){
+                  console.log('Error occurred while getting bib info');
+                } else {
+                  console.log('Error occurred while getting item info');
+                }
+                reject(errorResourceReq);
+              }
+            });
+        }
+      } else {
+            var entries = results.data.entries;
+            var entry = entries[0];
+            console.log(JSON.stringify(entry));
+            resolve(entry);
+          }
+  });
+}
 
-  var id = crypto.randomBytes(20).toString('hex').toString()
+//get wrapper access token
+var set_wrapper_access_token = function(){
+  return new Promise(function(resolve, reject){
+    wrapper.auth(function (error, authResults){
+      if(error){
+        console.log(error, error.stack);
+        reject ("Error occurred while getting access token");
+      }
+      resolve (wrapper.authorizedToken);
+    });
+  });
+}
 
+//send data to kinesis Stream
+var postResourcesToStream = function(resource, stream){
```

Comment:
Yes. I totally agree. So, as this was my first node app, I was shadowing how the discovery-queue-manager was written and its all in one index.js file. I guess this is an area that would be helpful for me to get some suggestions as to what the convention is. Do we create a new package and class to handle stream specific operations or just use the index.js file like how I have done here? I didn't delete the packages under lib because Matt had created them and wanted to hear from you guys as to whether use them or not.